pkg/app/app.go

```diff
@@ -159,3 +163,39 @@ func RealMain(opts types.Options, d types.Deployer, tester types.Tester) (result
 	}
 	return nil
 }
+
+func writeVersionToMetadataJSON(opts types.Options, d types.Deployer) error {
+	// setup the json metadata writer
+	metadataJSON, err := os.Create(
+		filepath.Join(opts.RunDir(), "metadata.json"),
+	)
```

Comment:
Any reason this can't match the behavior of `pkg.testers.metadata.WriteVersionToMetadata` and preserve an existing metadata if it happens to exist?

Reply:
It can, but atleast at this point kubetest2 itself doesn't expect any `metadata.json` to be written out to the run specific output directory by itself.